src/main/java/cheese/Parser/Parser.java

```diff
@@ -0,0 +1,229 @@
+package cheese.Parser;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+import cheese.Task.Task;
+
+/**
+ * Parser class to parse user input
+ */
+
+public class Parser {
+
+    /** Regex patterns for parsing user input */    
+    final static String byPatternString = "\\(by: (.*?)\\)";
+    final static String fromPatternString = "from: (.*?) to:";
+    final static String toPatternString = "to: (.*)\\)$";
+
+  final static Pattern byPattern = Pattern.compile(byPatternString);
+  final static Pattern fromPattern = Pattern.compile(fromPatternString);
+  final static Pattern toPattern = Pattern.compile(toPatternString);
```

Comment:
All above variables should be in Screaming Snake Case. It might also be better to directly input the strings into the methods to initiate the Patterns instead if the strings are only used once.